primitives/src/adapter.rs

```diff
@@ -33,14 +33,10 @@ impl fmt::Display for AdapterError {
     }
 }
 
-#[derive(Debug, Clone)]
-pub enum AdapterOptions {
-    DummAdapter {
-        dummy_identity: String,
-        dummy_auth: HashMap<String, String>,
-        dummy_auth_tokens: HashMap<String, String>,
-    },
-    EthereumAdapter(KeystoreOptions),
+pub struct DummyAdapterOptions {
```

Comment:
This could be moved to the `DummyAdapter` impl., as it's only needed there, for that specific implementation.